types.nix

```diff
@@ -1,9 +1,9 @@
 { lib }:
 with lib;
 with builtins;
-
-rec {
-
+let
+  safeZpoolName = pool: builtins.replaceStrings [ "-" ] [ "_" ] pool;
+in rec {
```

Comment:
hmm, I would prefer to abort on a invalid/unsafe zpool name rather than to replace it silently. replacing could lead to conflicts

Reply:
it is for specific purpose of creating an environment variable name, could make it clearer

Reply:
could they still conflict on similar zpool names? like my-pool and my_pool? maybe we don't care about that?

Reply:
this should solve it once and for good https://github.com/nix-community/disko/pull/112/commits/9eeafb271583e4fbe7c05580cf9debcb7554aea0

Reply:
I guess we use now tmpfiles for the pool names, not sure if this change is still relevant